legend-engine-test-runner-mapping/src/main/java/org/finos/legend/engine/testable/mapping/extension/MappingTestRunner.java

```diff
@@ -92,121 +96,88 @@ public TestResult executeAtomicTest(Root_meta_pure_test_AtomicTest atomicTest, P
+            // running of each test is catchable and put under a test error
+            for (MappingTest mappingTest : mappingTests)
             {
-                if (atomicTestIds.contains(mappingFunctionTest.id))
+                if (atomicTestIds.contains(mappingTest.id))
                 {
-                    org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult testResult = executeMappingFuncTest(mappingFunctionTest, runtime, context);
-                    testResult.testable = getElementFullPath(pureMapping, pureModel.getExecutionSupport());
-                    testResult.testSuiteId = mappingDataTestSuite.id;
+                    org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult testResult = executeMappingTest(mappingTest, context);
+                    testResult.testable = testablePath;
+                    testResult.testSuiteId = compiledMappingTestSuite._id();
                     results.add(testResult);
                 }
             }
-        }
-        catch (Exception exception)
-        {
-            throw new RuntimeException("Exception occurred executing service test suites.\n" + exception);
-        }
-        finally
-        {
-            this.closeConnections(connections);
-        }
-        return results;
-    }
 
-    public List<TestResult> executeMappingFuncTestSuite(MappingFunctionTestSuite mappingFunctionTestSuite, List<String> atomicTestIds, MappingTestRunnerContext context)
-    {
-        List<org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult> results = Lists.mutable.empty();
-        try
+        }
+        catch (Exception e)
         {
-            PureModel pureModel = context.getPureModel();
-            // build plan, executor args
-            LambdaFunction<?> pureLambda = HelperValueSpecificationBuilder.buildLambda(mappingFunctionTestSuite.func, new CompileContext.Builder(pureModel).build());
-            List<MappingDataTest> dataTests = this.validateMappingSuiteTests(mappingFunctionTestSuite, MappingDataTest.class);
-            Root_meta_pure_runtime_Runtime_Impl runtime = buildRuntimeForMultiInputTestSuite(dataTests, context);
-            SingleExecutionPlan executionPlan = PlanGenerator.generateExecutionPlan(pureLambda, pureMapping, runtime, null, context.getPureModel(), this.pureVersion, PlanPlatform.JAVA, null,
-                    context.routerExtensions, context.executionPlanTransformers);
-            context.getExecuteBuilder().withPlan(executionPlan);
-            for (MappingDataTest dataTest : dataTests)
+            // this is to catch any error for the setup of the test suite. we return test error for each test run
+            for (Root_meta_pure_test_AtomicTest testedError: compiledMappingTestSuite._tests())
             {
-                if (atomicTestIds.contains(dataTest.id))
+                if (atomicTestIds.contains(testedError._id()) && results.stream().noneMatch(t -> t.atomicTestId.equals(testedError._id())))
```

Comment:
this is wrong. We are saying testError even for tests which didn't run we should have try catch inside where we looping over tests to catch this correctly

Reply:
Actually looking again, we just catch test suite error here. For individual test we already send test error. If condition is misleading

Reply:
yeah my thinking here is if we fail on the setup of the suites we should throw testError for all running tests. We already catch errors for each indiviual tests. See comment on top of if function